File: python/2020/june/two-city-scheduling.py
from typing import List


class Solution:
    def two_cities(self, costs: List[List[int]]) -> int:
        refund = []
        n = len(costs) // 2
        min_cost = 0
        for a, b in costs:
            refund.append(b - a)
            min_cost += a
        refund.sort()
        for i in range(n):
            min_cost += refund[i]
        return min_cost

File: go/2020/june/two-city-scheduling.go
package june

import s "sort"

func twoCityShedCost(costs [][]int) int {
	var n = len(costs) / 2
	refund := make([]int, n*2)
	var minCost = 0
	var index = 0

	for _, c := range costs {
		refund[index] = c[1] - c[0]
		minCost += c[0]
	}
	s.Ints(refund)

	for i := 0; i < n; i++ {
		minCost += refund[i]
	}
	return minCost

}
